src/celengine/astrodb.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <unordered_map>
+#include "selection.h"
+#include "name.h"
+#include "astroobj.h"
+#include "astrocat.h"
+
+class AstroDatabase {
+public:
+    
+    typedef std::unordered_map<AstroCatalog::IndexNumber, AstroCatalog::IndexNumber> CrossIndex;
+    typedef std::unordered_map<AstroCatalog::IndexNumber, AstroObject*> MainIndex;
+    
+    enum Catalog
```

Comment:
`enum class Catalog`

Reply:
Ok, however I don't understand a difference...

Reply:
`class enum` is safe variant of `enum`, for example, this code will not compile:
```
class enum Foo {A, B, C};
void func(Foo);
...
func(1);
```

Reply:
Ok

Reply:
There is a problem with it. Adding `class` prevent usage of enumeration as array index (which break `static const char *CatalogPrefix[MaxBuiltinCatalog];` for example).